### PR TITLE
[docs] collapse navbar for speed

### PR DESCRIPTION
**Pull request by maxpumperla**
Fully collapses the navigation (meaning: every click on a nav item triggers a dynamic reload, no static drop-downs), as this turns out to be the best option for our needs. Drastically speed up our build, while fully keeping the new API section structure intact. The main drawback of this change is that we lose the visual clues telling you whether a nav item is shallow or nested (drops down).

On my local machine, I see the following build speeds (for reference, right now it's more than an hour):

```
Clean run:
make html  445.64s user 19.38s system 97% cpu 7:58.52 total


Incremental build:
make html  64.10s user 5.28s system 92% cpu 1:15.24 total
```

In other words, with this change we shouldn't have any problems with build times anymore. This is how the nav looks like at first, fully collapsed:

Screenshot 2023-03-28 at 14 03 42

Then, when you click through the nav tree, you see the navigation evolve as before this change:

Screenshot 2023-03-28 at 14 04 48

### CODE REVIEW

**Comment by richardliaw** [comment]
Can we still render the icon? Or else it's hard to know that it is a dropdown

**Comment by maxpumperla** [comment]
@richardliaw tried lots of things today, seems a bit complicated.

**Review by zhe-thoughts — APPROVED** [review]
Approved for merging into master (after we resolve @richardliaw 's comment on rendering)

**Review by richardliaw — APPROVED** [review]
